frontend/components/initiative/Editor.vue

```diff
@@ -0,0 +1,119 @@
+<script setup lang="ts">
+import { type EditorInitiative, isComplete } from '@/lib/editor'
+
+const props = defineProps<{
+  editorInitiative: EditorInitiative
+}>()
+
+const emit = defineEmits<(e: 'update:editorInitiative', ei: EditorInitiative) => void>()
+
+const model = computed({
+  get: () => props.editorInitiative,
+  set: (editorInitiative: EditorInitiative) => { emit('update:editorInitiative', editorInitiative) },
+})
+</script>
+
+<template>
+  <div>
+    <FormField
+      label="Initiative Name"
+      help-text="The name of the PACTA initiative."
+      :required="model.name.isRequired"
+      :completed="isComplete(model.name)"
+    >
+      <PVInputText
+        v-model="model.name.currentValue"
+      />
+    </FormField>
+    <FormField
+      label="Initiative ID"
+      help-text="This is the immutable unique identifier for the initiative. It can only contain alphanumeric characters, underscores, and dashes. This value will be shown in URLs, but will typically not be user visible."
+      :required="model.id.isRequired"
+      :completed="isComplete(model.id)"
```

Comment:
There's a lot of repetition of these two props on a bunch of `FormField`s, any reason we can't just do something like `:validate="model.id"` and have all the info about "is it required?" and "how do I validate it" baked into the `Editor*`? Then, it's one nice cohesive source of truth and less repetition.

Reply:
Great Idea! Done, after much struggle. This PR is WAY BETTER OFF for this comment.